app/models/manageiq/providers/nuage/manager_mixin.rb

```diff
@@ -2,24 +2,37 @@ module ManageIQ::Providers::Nuage::ManagerMixin
   extend ActiveSupport::Concern
 
   module ClassMethods
-    def raw_connect(auth_url, username, password)
-      ManageIQ::Providers::Nuage::NetworkManager::VsdClient.new(auth_url, username, password)
+    def raw_connect(username, password, endpoint_opts)
+      protocol    = endpoint_opts[:protocol].strip if endpoint_opts[:protocol]
+      hostname    = endpoint_opts[:hostname].strip
+      api_port    = endpoint_opts[:api_port]
+      # In case API port is represented as a string, ensure it has no whitespaces.
+      api_port.strip! if api_port.kind_of?(String)
+      api_version = endpoint_opts[:api_version].strip
+
+      url = auth_url(protocol, hostname, api_port, api_version)
+      _log.info("Connecting to Nuage VSD with url #{url}")
+      ManageIQ::Providers::Nuage::NetworkManager::VsdClient.new(url, username, password)
+    end
+
+    def auth_url(protocol, server, port, version)
+      scheme = protocol == "ssl-with-validation" ? "https" : "http"
+      "#{scheme}://#{server}:#{port}/nuage/api/#{version}"
```

Comment:
This should use a URI builder instead (though I see this is a copy, so it can be done in a separate PR).  The main reason is that for IPv6, if an IPv6 server address is given, this code won't actually work...prefer

```ruby
URI::Generic.build(:scheme => scheme, :host => server, :port => port, :path => "/nuage/api/#{version}")
```